operators/pkg/apis/common/v1alpha1/common.go

```diff
@@ -64,10 +64,18 @@ type TLSOptions struct {
 	Certificate SecretRef `json:"certificate,omitempty"`
 }
 
+// Enabled returns true when TLS is enabled based on this option struct.
+func (tls TLSOptions) Enabled() bool {
+	selfSigned := tls.SelfSignedCertificate
+	return selfSigned == nil || !selfSigned.Disabled || tls.Certificate.SecretName != ""
```

Comment:
Looks tricky enough to deserve a unit test maybe?

Reply:
added